### PR TITLE
Add LDoc and publishing to gh-pages

### PR DESCRIPTION
**Pull request by badosu**
See also https://github.com/beyond-all-reason/spring/discussions/549

### CODE REVIEW

**Comment by badosu** [comment]
I'm marking as ready for review now so it can be merged before it becomes untenable to rebase

**Comment by badosu** [comment]
There's still stuff to do in unsynced ctrl and it's missing both reads. It's also far from what I want for the final state.

As I move a few methods around so the code is more organized and follows the documentation sections rebases are going to become harder and harder so I want to put this out before that is the case.

**Comment by badosu** [comment]
I'll merge by end of weekend if there aren't any concerns.

File: rts/Lua/LuaUnsyncedCtrl.cpp [hunk 1]
```diff
@@ -3407,6 +4105,139 @@ int LuaUnsyncedCtrl::SetWindowMaximized(lua_State* L)
...
+ * @string text
+ * @treturn nil
+ */
+int LuaUnsyncedCtrl::SetClipboard(lua_State* L)
+{
+	SDL_SetClipboardText(luaL_checkstring(L, 1));
+	return 0;
+}
+
+
```
Comment (lhog) [inline rts/Lua/LuaUnsyncedCtrl.cpp:4240]: /*** Relinquish control of the game loading thread and OpenGL context back to the UI (LuaIntro) to draw screen updates and process windows events. Should be called after each widget/unsynced gadget is loaded in widget/gadget handler.
 *
 * @function Spring.Yield
 *
 * if this call returns true then caller should continue calling Spring.Yield during the widgets/gadgets load, if false - it shouldn't call it any longer.
 *
 * @number sleep time in milliseconds.
 * @treturn bool
 */

Comment (badosu) [reply rts/Lua/LuaUnsyncedCtrl.cpp:4240]:
```cpp
/*** Relinquish control of the game loading thread and OpenGL context back to the UI (LuaIntro) to draw screen updates and process windows events.
  *
   * @function Spring.Yield
   *  Should be called after each widget/unsynced gadget is loaded in widget/gadget handler.
   * @number sleep time in milliseconds.
   * @treturn bool if this call returns true then caller should continue calling Spring.Yield during the widgets/gadgets load, if false - it shouldn't call it any longer.
    */
```

**Comment by lhog** [comment]
It's been a bumpy day on the engine side, please update and I'll press the merge if it's ready.

**Comment by badosu** [comment]
> It's been a bumpy day on the engine side, please update and I'll press the merge if it's ready.

I'm resolving an engine issue unrelated to this at the moment, I'll get back to this when I can.

**Pull request merged**